service/history/timerQueueActiveProcessor.go

```diff
@@ -422,6 +449,79 @@ Update_History_Loop:
 	return ErrMaxAttemptsExceeded
 }
 
+func (t *timerQueueActiveProcessorImpl) processRetryTimer(task *persistence.TimerTaskInfo) error {
+	t.metricsClient.IncCounter(metrics.TimerTaskRetryTimerScope, metrics.TaskRequests)
+	sw := t.metricsClient.StartTimer(metrics.TimerTaskRetryTimerScope, metrics.TaskLatency)
+	defer sw.Stop()
+
+	processFn := func() error {
+		context, release, err0 := t.cache.getOrCreateWorkflowExecution(t.timerQueueProcessorBase.getDomainIDAndWorkflowExecution(task))
+		released := false
+		defer func() {
+			if !released {
+				release()
```

Comment:
You want to merge the change which takes release with an error, and update this PR with the correct release API.

Reply:
sure, I need to merge that change, as well as your change to the mutable state, there might be other conflicts, I want to hold on rebasing until this diff is good.